Marlin/src/pins/stm32f1/pins_MKS_ROBIN.h

```diff
@@ -101,117 +101,190 @@
+  #endif
+  
+  // MKS Robin TFT with ST7789V
+  //#define ST7789V
+  #if ENABLED(ST7789V)
+    #define TFT_DRIVER                      ST7789V
+    #define TFT_BUFFER_SIZE                 14400
+    #define ST7789V_COLOR_RGB
+
+    // YV for normal screen mounting
+    //#define ST7789V_ORIENTATION  ST7789V_MADCTL_MY | ST7789V_MADCTL_MV
+    // XV for 180° rotated screen mounting
+    #define ST7789V_ORIENTATION             ST7789V_MADCTL_MX | ST7789V_MADCTL_MV
+  #endif
 #endif
 
+//
+// SPI
+//
+#define ENABLE_SPI2
+#define SCK_PIN                             PB13
+#define MISO_PIN                            PB14
+#define MOSI_PIN                            PB15
+// SPI2 is already shared by LCD touch driver and flash
+// SPI2 MOSI (PB15) can be retrieved by soldering job on R17
+// SPI1(PA7) & SPI3(PB5) are not available
+
+
+//
+// Trinamic TMC2208/2209 UART
+//
 #if HAS_TMC_UART
-  /**
-   * TMC2208/TMC2209 stepper drivers
-   *
-   * Hardware serial communication ports.
-   * If undefined software serial is used according to the pins below
-   */
-  //#define X_HARDWARE_SERIAL  Serial1
-  //#define X2_HARDWARE_SERIAL Serial1
-  //#define Y_HARDWARE_SERIAL  Serial1
-  //#define Y2_HARDWARE_SERIAL Serial1
-  //#define Z_HARDWARE_SERIAL  Serial1
-  //#define Z2_HARDWARE_SERIAL Serial1
-  //#define E0_HARDWARE_SERIAL Serial1
-  //#define E1_HARDWARE_SERIAL Serial1
-  //#define E2_HARDWARE_SERIAL Serial1
-  //#define E3_HARDWARE_SERIAL Serial1
-  //#define E4_HARDWARE_SERIAL Serial1
-
-  // Unused servo pins may be repurposed with SoftwareSerialM
-  //#define X_SERIAL_TX_PIN                 PF8   // SERVO3_PIN -- XS2 - 6
-  //#define Y_SERIAL_TX_PIN                 PF9   // SERVO2_PIN -- XS2 - 5
-  //#define Z_SERIAL_TX_PIN                 PA1   // SERVO1_PIN -- XS1 - 6
-  //#define E0_SERIAL_TX_PIN                PC3   // SERVO0_PIN -- XS1 - 5
-  //#define X_SERIAL_RX_PIN      X_SERIAL_TX_PIN
-  //#define Y_SERIAL_RX_PIN      Y_SERIAL_TX_PIN
-  //#define Z_SERIAL_RX_PIN      Z_SERIAL_TX_PIN
-  //#define E0_SERIAL_RX_PIN    E0_SERIAL_TX_PIN
-
-  // Reduce baud rate for software serial reliability
-  #if HAS_TMC_SW_SERIAL
-    #define TMC_BAUD_RATE                  19200
+  // Hardware serial on Serial 1
+  #define TMC_HARDWARE_SERIAL
+  #if ENABLED(TMC_HARDWARE_SERIAL)
+    #define X_HARDWARE_SERIAL               Serial1
+    #define X2_HARDWARE_SERIAL              Serial1
+    #define Y_HARDWARE_SERIAL               Serial1
+    #define Y2_HARDWARE_SERIAL              Serial1
+    #define Z_HARDWARE_SERIAL               Serial1
+    #define Z2_HARDWARE_SERIAL              Serial1
+    #define E0_HARDWARE_SERIAL              Serial1
+    #define E1_HARDWARE_SERIAL              Serial1
+    #define E2_HARDWARE_SERIAL              Serial1
+    #define E3_HARDWARE_SERIAL              Serial1
+    #define E4_HARDWARE_SERIAL              Serial1
+  #endif
+
+  // Software serial on unused servo pins
+  #define TMC_SOFTWARE_SERIAL
```

Comment:
You are enabling both TMC_HARDWARE_SERIAL and TMC_SOFTWARE_SERIAL.

Reply:
My bad, i'll fix it in no time.